cli/src/cmd/forge/node.rs

```diff
@@ -0,0 +1,112 @@
+use clap::Parser;
+use ethers::{
+    core::{k256::ecdsa::SigningKey, types::U256},
+    prelude::{coins_bip39::English, MnemonicBuilder, Wallet},
+};
+
+use crate::opts::evm::EvmArgs;
+use foundry_node::NodeConfig;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Parser)]
+pub struct NodeArgs {
+    #[clap(flatten, next_help_heading = "EVM OPTIONS")]
+    pub evm_opts: EvmArgs,
+
+    #[clap(
+        long,
+        help = "either a comma-separated hex-encoded list of private keys, or a mnemonic phrase",
+        default_value = "10,test test test test test test test test test test test junk"
+    )]
+    pub accounts: SignerAccounts,
+
+    #[clap(long, help = "the balance of every genesis account")]
+    pub balance: Option<U256>,
+}
+
+impl NodeArgs {
+    pub fn into_node_config(self) -> NodeConfig {
+        let NodeConfig { chain_id, gas_limit, gas_price, genesis_balance, .. } =
+            NodeConfig::default();
+        NodeConfig::default()
```

Comment:
Additional config flags that would be useful to expose:
- `--hardfork`, `-H` to specify which EVM hardfork uses. Looks like this currently defaults to London, and not all EVM chains support the London EIPs
- `--port`, `-P` to change the port the node is exposed on. Seems this is currently a constant
- `--derivation-path` to change the derivation path and initial account index used by the provided mnemonic
- `--unlimited-contact-size` to enable/disable the contract size limit

Reply:
It would also be useful to be able to set the host, for example `0.0.0.0` in docker vs `localhost` when running locally. We currently maintain a dockerized hardhat node that we use to run tests against